library/ZendService/Twitter/Media.php

```diff
@@ -0,0 +1,251 @@
+
+        if (! $fileHandle) {
+            throw new Exception\MediaUploadException('Failed to open the file in the APPEND method.');
+        }
+
+        while (! feof($fileHandle)) {
+            $data = fread($fileHandle, $this->chunkSize);
+
+            $payload['media_data'] = base64_encode($data);
+            $payload['segment_index'] = $this->data['segment_index']++;
+
+            $httpClient->resetParameters();
+            $httpClient->setHeaders(['Content-type' => 'application/x-www-form-urlencoded']);
+            $httpClient->setMethod('POST');
+            $httpClient->setParameterPost($payload);
+
+            $response = $httpClient->send();
+            $appendStatus = $response->isSuccess();
+
+            if (! $appendStatus) {
+                throw new Exception\MediaUploadException(
+                    'Failed uploading segment '
+                    . ($this->data['segment_index'] - 1)
+                    . ' of '
+                    . $this->data['image_filename']
+                    . '. Error Code: ' . $response->getStatusCode()
+                    . '. Reason: ' . $response->getReasonPhrase()
+                );
+            }
+        }
+
+        fclose($fileHandle);
+
+        return $appendStatus;
+    }
+
+    /**
+     * Send the upload finalize to Twitter
+     *
+     * @param Http\Client $httpClient
+     * @params array $params
+     * @return Twitter\Response
+     */
+    protected function finalizeUpload($httpClient, array $params)
+    {
+        $payload = [
+            'command'  => 'FINALIZE',
+            'media_id' => $this->data['media_id'],
+        ];
+
+        $httpClient->resetParameters();
+        $httpClient->setHeaders(['Content-type' => 'application/x-www-form-urlencoded']);
+        $httpClient->setMethod('POST');
+        $httpClient->setParameterPost($payload);
+        $response = $httpClient->send();
+
+        return new Response($response);
+    }
+
+    /**
+     * Validate that the file exists and can be opened.
+     *
+     * @todo Put a check to make sure the file is local.
+     * @param string $fileName
+     * @return boolean
+     */
+    protected function validateFile($fileName)
+    {
+        $returnValue = false;
+        $returnValue = file_exists($fileName);
+
+        if ($returnValue) {
+            set_error_handler($this->createErrorHandler(), E_WARNING);
+            $returnValue = ($handle = fopen($fileName, 'rb'));
+            restore_error_handler();
+
+            if ($returnValue) {
+                fclose($handle);
+            }
+        }
+
+        return $returnValue;
+    }
+
+    /**
+     * Method overloading
+     *
+     * @param  string $key
+     * @return mixed
+     */
+    public function __get($key)
+    {
+        return isset($this->data[$key]) ? $this->data[$key] : null;
+    }
+
+    /**
+     * Method overloading
+     *
+     * @param string $key
+     * @param mixed $value
+     */
+    public function __set($key, $value)
```

Comment:
Absolutely not OK anymore. We used to do this in ZF1, but now this should be an endpoint per property, if mutability is wished. Otherwise, constructor should be preferred